planner/core/task.go

```diff
@@ -1748,9 +1750,8 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 	t := tasks[0].copy()
 	if cop, ok := t.(*copTask); ok {
 		// We should not push agg down across double read, since the data of second read is ordered by handle instead of index.
-		// The `extraHandleCol` is added if the double read needs to keep order. So we just use it to decided
-		// whether the following plan is double read with order reserved.
-		if cop.extraHandleCol != nil || len(cop.rootTaskConds) > 0 {
+		// We use doubleReadWithOrderReserved to decided whether the following plan is double read with order reserved.
+		if cop.doubleReadWithOrderReserved || len(cop.rootTaskConds) > 0 {
```

Comment:
Maybe I am too conservative on adding new fields to CopTask, but can we use `(cop.indexPlan != nil && cop.tablePlan != nil && cop.keepOrder)` to represent `doubleReadWithOrderReserved`, instead of adding this field to CopTask?

Reply:
Yeah, i think it works.